sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/resource/eks.py

```diff
@@ -0,0 +1,137 @@
+from urllib.request import Request, urlopen
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+from opentelemetry.semconv.resource import (
+    CloudPlatformValues,
+    CloudProviderValues,
+    ResourceAttributes,
+)
+
+logger = logging.getLogger(__name__)
+
+_CONTAINER_ID_LENGTH = 64
+_GET_METHOD = "GET"
+
+
+def _aws_http_request(method, path, cred_value, cert_data):
+    with urlopen(
+        Request(
+            "https://kubernetes.default.svc" + path,
+            headers={"Authorization": cred_value},
+            method=method,
+        ),
+        timeout=2000,
+        context=ssl.create_default_context(cadata=cert_data),
+    ) as response:
+        return response.read().decode("utf-8")
+
+
+def _get_k8s_cred_value():
+    try:
+        with open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/token",
+            encoding="utf8",
+        ) as token_file:
+            return "Bearer " + token_file.read()
+    # pylint: disable=broad-except
+    except Exception as exception:
+        logger.error("Failed to get k8s token: %s", exception)
+        raise exception
+
+
+def _is_eks(cred_value, cert_data):
+    return _aws_http_request(
+        _GET_METHOD,
+        "/api/v1/namespaces/kube-system/configmaps/aws-auth",
+        cred_value,
+        cert_data,
+    )
+
+
+def _get_cluster_info(cred_value, cert_data):
+    return _aws_http_request(
+        _GET_METHOD,
+        "/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info",
+        cred_value,
+        cert_data,
+    )
+
+
+def _get_cluster_name():
+    cred_value = _get_k8s_cred_value()
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    ) as cert_file:
+        k8_cert_data = cert_file.read()
+        if not _is_eks(cred_value, k8_cert_data):
+            return Resource.get_empty()
+
+    cluster_info = json.loads(_get_cluster_info(cred_value, k8_cert_data))
+    cluster_name = ""
+    try:
+        cluster_name = cluster_info["data"]["cluster.name"]
+    except KeyError as exception:
+        logger.warning("Cannot get cluster name on EKS: %s", exception)
+
+    return cluster_name
+
+
+def _get_container_id():
+    container_id = ""
+    with open("proc/self/cgroup", encoding="utf8") as container_info_file:
+        for raw_line in container_info_file.readlines():
+            line = raw_line.strip()
+            if len(line) > _CONTAINER_ID_LENGTH:
+                container_id = line[-_CONTAINER_ID_LENGTH:]
+    return container_id
+
+
+class AwsEksResourceDetector(ResourceDetector):
+    """Detects attribute values only available when the app is running on AWS
+    Elastic Kubernetes Service (EKS) and returns them in a Resource.
+    """
+
+    def detect(self) -> "Resource":
+        try:
+            cluster_name = _get_cluster_name()
+            container_id = _get_container_id()
+
+            if not container_id or not cluster_name:
+                return Resource.get_empty()
```

Comment:
Probably want to log a message here.

Reply:
Thanks for the catch! I think I'll just opt to use a `RuntimeError` and let the `except:` catch it instead:

```suggestion
                raise RuntimeError("Neither cluster name nor container ID found on EKS process.")
```